### ISSUE TITLE
type-compiler produces empty ops for some types and broken output

### ISSUE THREAD

**Issue by wielski** [deepkit/deepkit-framework#352]
type-compiler produces empty ops for type `Result` from https://github.com/traverse1984/oxide.ts

```
import { Result } from 'oxide.ts';
export type MyResult = Result<number, Error>;
```
causes broken output:
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.__ΩMyResult = void 0;
const __ΩMyResult; <--- empty
exports.__ΩMyResult = __ΩMyResult;
//# sourceMappingURL=index.js.map
```
with error `'Const declarations' require an initialization value.`

`Result` is both type and a class constructor.

It happens because of `extractPackStructOfTypeReference` calls for `extractPackStructOfType` with node `FunctionDeclaration`, that stops working because of `config.mode` is `never`.
I suspect it happens because of `findReflectionFromPath` reading `tsconfig.json` from oxide.ts inside node_modules folder, and returns never cause there is no reflection field.

**Comment by marcj** [comment]
Yeah oxide.ts was not built using the type-compiler, so all their types will be excluded. In your case this results in an error since excluded types are not handled as `any` or `unknown`. So, the fix is to emit `__ΩMyResult` as any (or unknown) bytecode.

**Comment by wielski** [comment]
Taking a types from third party npm modules would be helpful.
In my case I want to deserialize json with optionable fields, using interface like this:
```
import { Result } from 'oxide.ts';
interface MyOptionableStruct {
  myOptionableField: Option<number>;
}
```

**Comment by marcj** [comment]
Yeah that's not possible due to technical restrictions

**Comment by wielski** [comment]
@marcj thanks for your help :)
I'll try to rebuild desired types with type-compiler

**Comment by marcus-sa** [comment]
@wielski this will be possible when https://github.com/deepkit/deepkit-framework/pull/517 lands
